src/main/java/datawave/data/normalizer/AbstractGeometryNormalizer.java

```diff
@@ -175,4 +178,48 @@ private String adjustHexRangeBigInteger(String hexValue, String format, boolean
             return String.format(format, new BigInteger(hexValue, 16).subtract(BigInteger.ONE));
     }
     
+    /**
+     * This is used to determine if we have a valid geo hash (tier + position).
+     * 
+     * @param value
+     * @return true if valid
+     */
+    public boolean validHash(String value) {
+        try {
+            short tier = getTier(value);
+            if (validTier(tier) && validLength(tier, value)) {
+                return validPosition(tier, getPosition(value));
+            }
+        } catch (NumberFormatException e) {
+            // not a valid hex string in the first place
+        }
+        return false;
+    }
+    
+    public short getTier(String value) {
+        return Short.parseShort(value.substring(0, 2), 16);
+    }
+    
+    public long getPosition(String value) {
+        if (value.length() == 2) {
+            return 0;
+        }
+        return Long.parseLong(value.substring(2), 16);
+    }
+    
+    public boolean validTier(short tier) {
+        return tier >= 0 && tier <= 0x1f;
```

Comment:
This is fine, but if we ever decide to change the 'indexStrategy' used by PointNormalizer and GeometryNormalizer, we may need to update these checks to behave differently.  Might be worth just adding a comment here, or on the line where the indexStrategy is defined so we don't forget.

Reply:
oh?  I did not realized this.  I will add a very large bold comment in this regard